cloud/gce/main.tf

```diff
@@ -32,30 +32,77 @@ resource "google_compute_instance" "cockroach" {
     key_file = "~/.ssh/${var.key_name}"
   }
 
+  service_account {
+    scopes = ["https://www.googleapis.com/auth/compute.readonly"]
+  }
+}
+
+# Generate contents of config.sh from its template file.
+#
+# TODO(cdo): support secure clusters
+resource "template_file" "config" {
+  count = "${var.num_instances}"
+
+  template = "${file("config.sh.tpl")}"
+  vars {
+    sql_port = "${var.sql_port}"
+    http_port = "${var.http_port}"
+    load_balancer_address = "${var.load_balancer_address}"
+    local_address = "${element(google_compute_instance.cockroach.*.network_interface.0.address, count.index)}"
+    # The value of the --join flag must be empty for the first node,
+    # and a running node for all others. We built a list of addresses
+    # shifted by one (first element is empty), then take the value at index "instance.index".
+    join_address = "${element(concat(split(",", ""), google_compute_instance.cockroach.*.network_interface.0.address), count.index)}"
+  }
+}
+
+resource "null_resource" "cockroach-runner" {
+  count = "${var.num_instances}"
+
+  connection {
+    user = "ubuntu"
+    key_file = "~/.ssh/${var.key_name}"
+    host = "${element(google_compute_instance.cockroach.*.network_interface.0.access_config.0.assigned_nat_ip, count.index)}"
+  }
+
+  triggers {
+    instance_ids = "${element(google_compute_instance.cockroach.*.id, count.index)}"
+  }
+
   provisioner "file" {
-    source = "${var.cockroach_binary}"
-    destination = "/home/ubuntu/cockroach"
+    source = "download_binary.sh"
+    destination = "/home/ubuntu/download_binary.sh"
   }
 
   provisioner "file" {
     source = "launch.sh"
     destination = "/home/ubuntu/launch.sh"
   }
 
+  provisioner "file" {
```

Comment:
That's not an argument because the only reason we don't do it is because we have no use for them. This creates a use and `./build/builder.sh make build` really isn't that hard.